File: my-submissions/e125.py
# https://leetcode.com/problems/valid-palindrome/description/

class Solution:
    def isPalindrome(self, s: str) -> bool:
        singleStr = re.sub('[^A-Za-z0-9]', '', s).lower()

        for i in range(len(singleStr) // 2) :
            if not singleStr[i] == singleStr[len(singleStr) - i - 1] :
                return False 

        return True

File: my-submissions/e125.c
bool isPalindrome(char* s) {
    int left = 0;
    int right = 0;

    while (s[right + 1]) {
        right++;
    }

    while (left < right) {
        if (s[left] <= 90 && s[left] >= 65) {
            s[left] += 32;
        } else if (!((s[left] <= 57 && s[left] >= 48) || s[left] <= 122 && s[left] >= 97)) {
            left++;
            continue;
        }

        if (s[right] <= 90 && s[right] >= 65) {
            s[right] += 32;
        } else if (!((s[right] <= 57 && s[right] >= 48) || (s[right] <= 122 && s[right] >= 97))) {
            right--;
            continue;
        }
        if (s[left] != s[right]) {
            return false;
        }
        left++;
        right--;
    }

    return true;
}